tcpdf.php

```diff
@@ -7866,6 +7892,60 @@ public function _destroy($destroyall=false, $preserve_objcopy=false) {
 		}
 	}
 
+	/** Page cache reference counts */
+	protected static $pageCacheRefCnts = array();
```

Comment:
can this be non static ?

Reply:
I intentionally have it static so that it can be shared when you clone it. The PR implements copy-on-write, so if you clone the PDF, it will share the cache until an edit is made.

I've been including this patch on production since 2019 and I haven't noticed any issues with it, so I feel like it's safer to leave this as-is, but if you want, I can remove the shared cache pages and make this non-static.

Reply:
I think it could cause issues to users that do not expect things to be cached between pdfs

Reply:
Do you mean people using the library? They would have to enable the page cache with `usePageCacheFile(...)` and even if they do, I have the code internally handling the caching between PDFs, so I don't think it would cause issues or be something users even know about.

If you are referring to people developing the library, that's where I think it might cause confusion, but I think I can handle that by writing more detailed comments.

If the `static` part is something you want to change, but still keep caching across PDFs, I could implement that with a new object for reference counting and since objects are shared by reference, it would act very similarly to the static variable.

Reply:
I think that if you can implement a version with objects that would be better than a static variable. @MauricioFauth maybe you would have input on this feature and the use of statics?

Reply:
I added e5a3a7ce85c5c18066fcff7f87c3cf3f25fca2ad, which does it with an object instead. I'm not as confident in this change since it hasn't had years of real world testing, but it seems correct.  Personally, I think this might be a little more confusing for library developers. With the static variable, I think it's more obvious that it's used to share data than this object by reference approach.

Reply:
Thanks, I hope someone with merge power can review this some time soon
The commit can be reverted if the static approach is preferred by the reviewer